### PR TITLE
Opensource mqtt bridge and webhook

### PR DESCRIPTION
**Pull request by zmstone**
All changes are already reviewed in previous PRs.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

File: apps/emqx_web_hook/etc/emqx_web_hook.conf [hunk 1]
```diff
@@ -5,10 +5,10 @@
 ## Webhook URL
 ##
 ## Value: String
-web.hook.url = http://127.0.0.1:80
+web.hook.api.url = http://127.0.0.1:80
```
Comment (tigercl) [inline apps/emqx_web_hook/etc/emqx_web_hook.conf:8]: web.hook.url

File: apps/emqx_web_hook/src/emqx_web_hook_actions.erl [hunk 2]
```diff
@@ -17,95 +17,111 @@
...
+
+-type(action_fun() :: fun((Data :: map(), Envs :: map()) -> Result :: any())).
+
+-type(url() :: binary()).
 
 -define(RESOURCE_TYPE_WEBHOOK, 'web_hook').
 -define(RESOURCE_CONFIG_SPEC, #{
-            url => #{
-                order => 1,
+    method => #{order => 1,
```
Comment (tigercl) [inline apps/emqx_web_hook/src/emqx_web_hook_actions.erl:41]: Delete `method` field, existed in Action's params

File: apps/emqx_web_hook/src/emqx_web_hook_actions.erl [hunk 3]
```diff
@@ -17,95 +17,111 @@
...
+                                 zh => <<"请求方法"/utf8>>}},
+    url => #{order => 2,
+             type => string,
+             format => url,
+             required => true,
+             title => #{en => <<"Request URL">>,
+                        zh => <<"请求 URL"/utf8>>},
+             description => #{en => <<"The URL of the server that will receive the Webhook requests.">>,
+                              zh => <<"用于接收 Webhook 请求的服务器的 URL。"/utf8>>}},
+    headers => #{order => 3,
```
Comment (tigercl) [inline apps/emqx_web_hook/src/emqx_web_hook_actions.erl:57]: Delete headers field

File: apps/emqx_web_hook/test/prop_webhook_confs.erl [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+%%--------------------------------------------------------------------
+%% Properties
+%%--------------------------------------------------------------------
+
+prop_confs() ->
+    Schema = cuttlefish_schema:files(filelib:wildcard(code:priv_dir(emqx_web_hook) ++ "/*.schema")),
+    ?ALL({Url, Confs0}, {url(), confs()},
+        begin
+            Confs = [{"web.hook.api.url", Url}|Confs0],
```
Comment (tigercl) [inline apps/emqx_web_hook/test/prop_webhook_confs.erl:39]: web.hook.api.url to web.hook.url

**Review by tigercl — DISMISSED** [review]

File: apps/emqx_web_hook/src/emqx_web_hook_actions.erl [hunk 5]
```diff
@@ -350,24 +334,27 @@ pool_opts(Params = #{<<"url">> := URL}, ResId) ->
     PoolSize = maps:get(<<"pool_size">>, Params, 32),
     ConnectTimeout = cuttlefish_duration:parse(str(maps:get(<<"connect_timeout">>, Params, <<"5s">>))),
     {Inet, Host} = parse_host(Host0),
+    SslOpts0 = maybe_ssl(Params, ResId, add_default_scheme(URL)),
+    TranOpts = lists:keyfind(transport_opts, 1, SslOpts0),
+    SslOpts = lists:keyreplace(transport_opts,  1,
```
Comment (zmstone) [inline apps/emqx_web_hook/src/emqx_web_hook_actions.erl]:
```
MoreOpts =
    case add_default_scheme(URL) of
        <<"https://", _/binary>> -> [{transport, ssl}, {transport_opts, [Inet | get_ssl_opts(Config, ResId)]}];
        _ -> [{transport_opts, [Inet]}]
end,
```
we can delete `maybe_ssl`

File: apps/emqx_web_hook/src/emqx_web_hook.appup.src [hunk 6]
```diff
@@ -0,0 +1,10 @@
+%% -*-: erlang -*-
+
+{VSN,
+  [
+    {<<".*">>, []}
+  ],
+  [
+    {<<".*">>, []}
```
Comment (terry-xiaoyu) [inline apps/emqx_web_hook/src/emqx_web_hook.appup.src:8]: `<<"*.">>` or `<<".*">>`, is this intentional？

File: apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt.appup.src [hunk 7]
```diff
@@ -0,0 +1,10 @@
+%% -*-: erlang -*-
+
+{VSN,
+  [
+    {<<".*">>, []}
+  ],
+  [
+    {<<"*.">>, []}
```
Comment (terry-xiaoyu) [inline apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt.appup.src:8]: `<<"*.">>` or `<<".*">>`, is this intentional？

**Review by terry-xiaoyu — APPROVED** [review]

**Review by terry-xiaoyu — DISMISSED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**